### PR TITLE
Adding setup_fedora.sh

### PR DESCRIPTION
**Pull request by storymode7**

### CODE REVIEW

File: setup/setup_fedora.sh [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+sudo dnf -y upgrade
+
+# Install zsh
+sudo dnf -y install zsh
+
+# Install everything else
+sudo dnf -y install ack
+sudo dnf -y install git git-extras hub 
+sudo dnf -y install htop ngrep nmap
+sudo dnf -y install autojump
```
Comment (CuriousLearner) [inline setup/setup_fedora.sh]: What is autojump?

**Review by CuriousLearner — COMMENTED** [review]
Please add comments to the alternatives; and also to stuff like `autojump`, `zopfli`

**Comment by storymode7** [comment]
Added comments and also ran a syntax check with ShellCheck, please review.

File: setup/setup_fedora.sh [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+sudo dnf -y install teamviewer
+
+###############################################################################
+#                        Install utilities                                    #
+###############################################################################
+
+sudo dnf -y install firefox
+#Evernote,skitch,dash not available
+sudo dnf -y install https://prerelease.keybase.io/keybase_amd64.rpm
+#android-file-transfer-linux http://whoozle.github.io/android-file-transfer-linux/
```
Comment (CuriousLearner) [inline setup/setup_fedora.sh:37]: Please use a space after `#`, that makes it more readable and for text editors to easily comment/un-comment lines.

File: setup/setup_fedora.sh [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+npm i -g postcss-cli
+npm i -g autoprefixer
+
+#Custom stuff
+#Telegram
+sudo dnf -y copr enable rommon/telegram
+sudo dnf -y install telegram-desktop
+sudo dnf copr disable rommon/telegram
+
+# Slack needs to be unpacked manually due to version name in download url
```
Comment (CuriousLearner) [inline setup/setup_fedora.sh:85]: I think we can include a particular version here and update it later, when needed. Does it have some sort of problem with it?

File: setup/setup_fedora.sh [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+###############################################################################
+
+set -P
+# Install Zsh settings
+ln -sf "$PWD/zsh/themes/curiouslearner.zsh-theme" "$HOME/.oh-my-zsh/themes"
+# Zsh Syntax highlighting
+git clone https://github.com/zsh-users/zsh-syntax-highlighting.git "${ZSH_CUSTOM:-~/.oh-my-zsh/custom}/plugins/zsh-syntax-highlighting"
+
+# Install Powerlevel9k theme
+git clone https://github.com/bhilburn/powerlevel9k.git ~/.oh-my-zsh/custom/themes/powerlevel9k
```
Comment (CuriousLearner) [inline setup/setup_fedora.sh:180]: For the powerline theme, did you installed the patch fonts anywhere?

How does your terminal looks after setup?

**Review by CuriousLearner — COMMENTED** [review]
Looks good overall. 

Just minor comments.

File: setup/setup_fedora.sh (hunk 4 above)
Comment (storymode7) [reply setup/setup_fedora.sh:180]: https://files.slack.com/files-pri/T0X8CRH5K-F8J9JMWBG/screenshot_from_2017-12-20_20-10-19.png

Comment (storymode7) [reply setup/setup_fedora.sh:180]: Yups, I installed patch fonts.

File: setup/setup_fedora.sh (hunk 3 above)
Comment (storymode7) [reply setup/setup_fedora.sh:85]: No nothing. Will update.

**Pull request closed without merging**